tests/errors.spec.ts

```diff
@@ -0,0 +1,43 @@
+/// <reference path="mochainon.d.ts" />
```

Comment:
Generally you never ever want to use `///<reference...`.

It's a leftover from before TypeScript had a proper module system. You either want to have an actual dependency TypeScript can understand and follow to get the type (e.g. a normal `import`), or you want to include the type definition in your `tsconfig.json`. Here the problem is just that your tsconfig.json only matches `lib/**/*.d.ts`, and it also needs `test/**/*d.ts` (or you could have a single top-level `typings` folder).

It's a little weird that this is `lib` and `test` together - in principle you could have test and prod tsconfigs, but I don't think it's worth it personally.